tern/analyze/common.py

```diff
@@ -452,3 +453,76 @@ def get_os_style(image_layer, binary):
                     package_manager=binary,
                     package_format=image_layer.pkg_format,
                     os_list=image_layer.os_guess), 'info'))
+
+
+def check_git_src(dockerfile_path, src_path):
+    '''Given the src_path and the dockerfile path, return the git
+    repository name and sha information in the format of string.'''
+    # get the path of the folder containing the dockerfile
+    dockerfile_folder_path = os.path.dirname(dockerfile_path)
+    # get the path of the target folder or file
+    src_path = os.path.join(dockerfile_folder_path, src_path)
+    logger.debug('looking into path: %s for git repo.', src_path)
+    comment_line = ''
+    if os.path.exists(src_path):
+        # if src is file, working dir should be the upper directory.
+        if not os.path.isdir(src_path):
+            src_path = os.path.dirname(src_path)
+        # locate the top level directory
+        path_to_toplevel = get_git_toplevel(src_path)
+        # path is not None
+        if os.path.exists(path_to_toplevel):
+            sha_info = get_git_sha(path_to_toplevel)
+            name_info = get_git_projcet_name(path_to_toplevel)
+            comment_line = ('git project name: ' + name_info +
+                            ', HEAD sha: ' + sha_info)
+        else:
+            comment_line = 'Not a git repository'
+    else:
+        comment_line = 'Unable to find the src file or foler'
+    return comment_line
+
+
+def get_git_sha(path_to_toplevel):
+    '''Given a absolute path to a git repository, return the HEAD sha.'''
+    command = ['git', 'rev-parse', 'HEAD']
+    sha_info = '(not found)'
+    try:
+        output = subprocess.check_output(  # nosec
+            command, stderr=subprocess.DEVNULL, cwd=path_to_toplevel)
+        if isinstance(output, bytes):
+            sha_info = output.decode('utf-8').split('\n').pop(0)
+    except subprocess.CalledProcessError:
+        logger.error("Cannot find git repo sha, toplevel path is %s",
+                     path_to_toplevel)
+    return sha_info
+
+
+def get_git_projcet_name(path_to_toplevel):
+    '''Given a absolute path to a git repository, return the project name.'''
+    command = ['basename', path_to_toplevel]
+    name_info = '(not found)'
+    try:
+        output = subprocess.check_output(  # nosec
+            command, stderr=subprocess.DEVNULL, cwd=path_to_toplevel)
+        if isinstance(output, bytes):
+            name_info = output.decode('utf-8').split('\n').pop(0)
+    except subprocess.CalledProcessError:
+        logger.error("Cannot find git repo name, toplevel path is %s",
+                     path_to_toplevel)
+    return name_info
+
+
+def get_git_toplevel(src_path):
+    '''Give a path, return true if it is in a git repository and return the
```

Comment:
Small nit: This description says the function returns true and also returns the absolute path to the top level directory when only the latter is true :)